### DISCUSSION TITLE
this.props.dangerousAsPath in _documents different on SSR and SSG

### DISCUSSION THREAD

**Discussion by hk86** [vercel/next.js#16046] [Help] [answered]
Hello there,

my app uses SSR and SSG. I have a multilingual website therefore I set  alternate link tags in the head tag at `_document.js`. In order to get the current path I use `this.props.dangerousAsPath`. From what I see it behaves different on SSR and SSG, if I deploy the app to Vercel. I'm not able to reproduce it locally, even if I run build.

i.e.: for the page `/pages/article/[id].js` SSG stores _/article/123_ in `this.props.dangerousAsPath`, but in case of SSR it stores `/article/123?id=123`. I see this behavior on the Vercel platform, not locally.

As you can see it adds a query string of the variables used in the path. Any ideas why this is the case?

**Comment by ijjk** [comment, marked as answer]
Hi, it sounds like you are running into [this issue](https://github.com/vercel/next.js/issues/15879) which is fixed in the latest canary of Next.js `v9.5.2-canary.13`. Please try upgrading to that version and let us know whether you are still experiencing that behavior or not

**Comment by hk86** [comment]
Yes! This fixed my issue. Thank you!
